src/vivarium/framework/results/observation.py

```diff
@@ -60,7 +59,8 @@ class BaseObservation(ABC):
     DataFrame or one with a complete set of stratifications as the index and
     all values set to 0.0."""
     results_gatherer: Callable[
-        [pd.DataFrame | DataFrameGroupBy[str], tuple[str, ...] | None], pd.DataFrame
```

Comment:
I'm assuming the things in this file were just type-hinted incorrectly?

Reply:
Yeah, though I'm not gonna lie - I'm having trouble understanding how to type-hint DataFrameGroupBys...